mobile/src/test/java/org/openhab/habdroid/model/OpenHAB2WidgetTest.java

```diff
@@ -1,67 +1,404 @@
 package org.openhab.habdroid.model;
 
+import net.bytebuddy.implementation.bytecode.Throw;
+
 import org.json.JSONObject;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.runners.MockitoJUnitRunner;
 
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.concurrent.ExecutionException;
+
 import static junit.framework.Assert.assertEquals;
 import static junit.framework.Assert.assertTrue;
 import static org.mockito.Mockito.mock;
 
 @RunWith(MockitoJUnitRunner.class)
 public class OpenHAB2WidgetTest {
-    OpenHABWidget rootWidget = mock(OpenHAB2Widget.class);
+    private OpenHABWidget rootWidget = mock(OpenHAB2Widget.class);
+    OpenHABWidget sut1;
```

Comment:
I like descriptive names, even for tests :P

Reply:
Descriptive names are better, but I cannot find one for these oh widgets. sutWithWidgetidDemoAndTypeGroupAnd... isn't good.

Reply:
Hmm, you're right, I'd some ideas when I read the code the first time, but forgot them :/ It's ok for now I think :)